### DISCUSSION TITLE
Assert\Type constraint on natively typed property : is it useless ?

### DISCUSSION THREAD

**Discussion by gnutix** [symfony/symfony#54330] [Q&A] [answered]
Hello there,

Is there any reason to have an `Assert\Type` attribute/annotation over a natively typed property ? From my testing, I've seen no difference in behavior with or without it.

I'm surprised there's no mention about this in the documentation, as [I'm not the first one to wonder](https://stackoverflow.com/questions/77097425/symfony-php-8-assert-typestring-is-really-useful).

Thanks for your insights.
gnutix

**Comment by MatTheCat** [comment, marked as answer]
General Kenobi :robot: 

There are only two behaviors when setting a value to a differently-typed property: either it crashes, or the value is silently converted to the property type (`false` being converted to an empty string e.g.).

That means the only case where the property is assigned, you’re guaranteed its value has the correct type. As such, the `Type` constraint won’t serve any purpose.

**Reply by gnutix** [reply]
Thanks for the confirmation!

**Reply by MatTheCat** [reply]
Late thought: you could use the constraint to assert a value has a subtype of its propertyʼs. Not sure if it is useful in practice 🤔

**Reply by gnutix** [reply]
We use it to assert lists of objects for example. Something like `#[AssertAll(new AssertType(SomeObject::class))]`.
